packages/server-wallet/jest/custom-matchers.ts

```diff
@@ -50,4 +51,44 @@ expect.extend({
       };
     }
   },
-});
\ No newline at end of file
+});
+
+expect.extend({
+  toBeObjectiveDoneType: async (
+    received: ObjectiveResult[],
+    expected: ObjectiveDoneResult['type']
+  ) => {
```

Comment:
Is there a different API that's more useful to the developer?

For instance, I might pass in an array `received: ObjectiveResult[]` and `expected: DeepPartial<ObjectiveDoneResult>[]`, enabling a developer to do

`await expect(wallet.createChannels(SNIP)).toBeObjectiveDoneType([{type: 'Succeed'}, {type: 'Fail', reason: 'Ledger is out of funds'}])` to test the behaviour of `createChannels`.

Reply:
That's a good point! 

I think my preference would be to wait until we need something like that and then modify/extend the custom matcher.

Reply:
How much work is it to modify it at this stage?

Reply:
I don't think it would be a lot of work!

I do think it will make these tests slightly less succinct:
```
 await expect(response).toBeObjectiveDoneType(Array(10).fill({type: 'Success'}))
// vs
  await expect(response).toBeObjectiveDoneType('Success');
```

Reply:
Ahhh, the current API checks that they all resolve to success, good point.